src/modules/jackrack/plugin_mgr.c

```diff
@@ -617,4 +695,393 @@ lv2_plugin_desc_t *lv2_mgr_get_any_desc(lv2_mgr_t *plugin_mgr, char *id)
+                              dlerror());
+          } */
+}
+
+
+static gint vst2_mgr_sort(gconstpointer a, gconstpointer b)
+{
+    const vst2_plugin_desc_t *da;
+    const vst2_plugin_desc_t *db;
+    da = (const vst2_plugin_desc_t *) a;
+    db = (const vst2_plugin_desc_t *) b;
+
+    return strcasecmp(da->name, db->name);
+}
+
+static void vst2_mgr_get_dir_plugins(vst2_mgr_t *vst2_mgr, const char *dir)
+{
+    DIR *dir_stream;
+    struct dirent *dir_entry;
+    char *file_name;
+    int err;
+    size_t dirlen;
+
+    dir_stream = opendir(dir);
+    if (!dir_stream) {
+        /*      mlt_log_warning( NULL, "%s: error opening directory '%s': %s\n",
+               __FUNCTION__, dir, strerror (errno)); */
+        return;
+    }
+
+    dirlen = strlen(dir);
+
+    while ((dir_entry = readdir(dir_stream))) {
+        struct stat info;
+
+        if (strcmp(dir_entry->d_name, ".") == 0
+            || mlt_properties_get(vst2_mgr->blacklist, dir_entry->d_name)
+            || strcmp(dir_entry->d_name, "..") == 0)
+            continue;
+
+        file_name = g_malloc(dirlen + 1 + strlen(dir_entry->d_name) + 1);
+
+        strcpy(file_name, dir);
+        if (file_name[dirlen - 1] == '/')
+            strcpy(file_name + dirlen, dir_entry->d_name);
+        else {
+            file_name[dirlen] = '/';
+            strcpy(file_name + dirlen + 1, dir_entry->d_name);
+        }
+
+        stat(file_name, &info);
+        if (S_ISDIR(info.st_mode))
+	  {
+	    vst2_mgr_get_dir_plugins(vst2_mgr, file_name);
+	  }
+        else {
+            char *ext = strrchr(file_name, '.');
+            if (ext
+                && (strcmp(ext, ".so") == 0 || strcmp(ext, ".dll") == 0
+                    || strcmp(ext, ".dylib") == 0)) {
+                vst2_mgr_get_object_file_plugins(vst2_mgr, file_name);
+            }
+        }
+
+        g_free(file_name);
+    }
+
+    err = closedir(dir_stream);
+    if (err)
+        mlt_log_warning(NULL,
+                        "%s: error closing directory '%s': %s\n",
+                        __FUNCTION__,
+                        dir,
+                        strerror(errno));
+}
+
+static void vst2_mgr_get_path_plugins(vst2_mgr_t *vst2_mgr)
+{
+    char *vst_path, *dir;
+
+    vst_path = g_strdup(getenv("VST_PATH"));
+#ifdef _WIN32
+    if (!vst_path) {
+        vst_path = malloc(strlen(mlt_environment("MLT_APPDIR")) + strlen("\\lib\\vst") + 1);
+        strcpy(vst_path, mlt_environment("MLT_APPDIR"));
+        strcat(vst_path, "\\lib\\vst");
+    }
+#elif defined(RELOCATABLE)
+#ifdef __APPLE__
+#define VST2_SUBDIR "/PlugIns/vst"
+#else
+#define VST2_SUBDIR "/lib/vst"
+#endif
+    {
+        vst_path = malloc(strlen(mlt_environment("MLT_APPDIR")) + strlen(VST2_SUBDIR) + 1);
+        strcpy(vst_path, mlt_environment("MLT_APPDIR"));
+        strcat(vst_path, VST2_SUBDIR);
+    }
+#else
+    if (!vst_path)
+        vst_path = g_strdup("/usr/local/lib/vst:/usr/lib/vst:/usr/lib64/vst");
```

Comment:
According to [Steinberg](https://steinbergmedia.github.io/vst3_dev_portal/pages/Technical+Documentation/Locations+Format/Plugin+Locations.html#on-linux-platform) the folder should be `vst3`, but I guess this only supports version 2, and I do not know the complications of that. Is it OK to include VST2 and VST3 together if some applications do not support 3? I downloaded the SwankyAmp you suggested, and it says `SwankyAmp.vst3`.

Reply:
No I don't think that is possible my code is designed to run vst2 plugins only I don't know for sure but I think initialization of vst3 plugin is different from vst2 so I can't actually if this is the case.

The version I downloaded of SwankyAmp is from arch GNU/Linux distrho-ports-vst packages (https://github.com/DISTRHO/DISTRHO-Ports) which contain the vst2 version.

It seems that I need more testing I try it with other plugins and the output does not match what audacity or ardour generate.

Reply:
OK, now I understand the situation better. Sounds like someone took the GPL SwankyAmp code and ported it to VST2  and Linux. No change required here.